DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/mmhost/README.md

```diff
@@ -0,0 +1,335 @@
+| `stable`                  | Guarantee that the pointer will not change between pipelined invocations of the kernel
+
+
+You can use the following parameters to configure your IP component's Avalon memory-mapped host interfaces:
+
+| Parameter                | Default Value | Description
+|---                       |---            |---
+|  `buffer_location<id>`   | N/A           | The address space of the interface that associates with the host. Each unique buffer location will result in a unique Avalon memory-mapped host interface. When `buffer_location` is not specified, then corresponding LSUs will connect to all of the IP's Avalon memory-mapped host interfaces.
+| `awidth<width>`          | 41            | Width of the Avalon memory-mapped host address bus in bits. 
+| `dwidth<width>`          | 64            | Width of the Avalon memory-mapped host data bus in bits. 
+| `latency<value>`         | 0             | Read latency for downstream fixed-latency Avalon memory-mapped agent. For a variable-latency Avalon memory-mapped agent, set `latency<0>`. 
+| `read_write_mode<mode>`  | `read_write`  | Port direction of the interface. (`read_write`, `read` or `write`) 
+| `maxburst<value>`        | 1             | Maximum number of data transfers that can associate with a read or write request. 
+| `alignment<alignment>`   | 1          | Alignment of the Avalon memory-mapped host interface
+
+We can use some of these parameters to improve the performance of `Example 1`. Re-structure the design so that each pointer had exclusive access to a dedicated Avalon memory-mapped agent memory, like this:
+
+![](assets/dedicated_avhost.svg)
+
+#### Example 3: A kernel with dedicated Avalon memory-mapped host interfaces
+```c++
+constexpr int kBL1 = 1;
+constexpr int kBL2 = 2;
+constexpr int kBL3 = 3;
+
+struct MultiMMIP{
+
+  annotated_ptr<int, decltype(properties{
+          buffer_location<kBL1>
+          })> x; 
+  annotated_ptr<int, decltype(properties{
+          buffer_location<kBL2>
+          })> y;
+  annotated_ptr<int, decltype(properties{
+          buffer_location<kBL3>
+          })> z;  
+  int size;
+
+  void operator()() const {
+    for (int i = 0; i < size; ++i) {
+      int mul = x[i] * y[i];
+      int add = mul + z[i];
+      y[i] = mul;
+      z[i] = add;
+    }
+  }
+};
+```
+
+### Interfacing with off-chip memory
+
+Most of the time, a component's interfaces will be subject to system constraints. Assume that the input and output vectors are too large for on-chip memory. Instead, assume a memory system with the following constraints:
+
+* Two banks of DDR
+* Data bus of 256 bits
+* bursts of up to 8 requests
+
+![](assets/ddr_avhost.svg)
+
+We can make better use of the available DDR bandwidth by coalescing the 32-bit wide load-store units into wider 256-bit load-store units to match the memory interface.
+
+#### Example 4: A kernel that interfaces with two off-chip memories
```

Comment:
Doesn't match your tutorial cpp file. Make them align and check that for all examples/files.